### PR TITLE
Set Xpress specific parameters before generic ones

### PR DESCRIPTION
**Pull request by tbittar**
Set specific parameters before generic parameters in Xpress interface, closes https://github.com/rte-france/or-tools/issues/127

### CODE REVIEW

File: ortools/linear_solver/xpress_interface.cc [hunk 1]
```diff
@@ -1830,13 +1830,14 @@ MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
...
   // Set parameters.
-  // NOTE: We must invoke SetSolverSpecificParametersAsString() _first_.
-  //       Its current implementation invokes ReadParameterFile() which in
-  //       turn invokes XPRSreadcopyparam(). The latter will _overwrite_
-  //       all current parameter settings in the environment.
+  // We first set our internal MPSolverParameters from 'param' and then set
+  // any user-specified internal solver parameters via
+  // solver_specific_parameter_string_.
+  // Default MPSolverParameters can override custom parameters (for example for
+  // presolving) and therefore we apply MPSolverParameters first.
```
Comment (pet-mit) [inline ortools/linear_solver/xpress_interface.cc:1837]:
```suggestion
  // Default MPSolverParameters can override custom parameters while specific
  // parameters allow a higher level of customization (for example for
  // presolving) and therefore we apply MPSolverParameters first.
```

**Review by pet-mit — CHANGES_REQUESTED** [review]
looks good to me
unfortunately we have to wait for some sync issues with google's main (after google merges my PR)

File: ortools/linear_solver/xpress_interface_test.cc [hunk 2]
```diff
@@ -766,6 +766,14 @@ TEST_F(XpressFixtureLP, SetPrimalTolerance) {
   EXPECT_EQ(getter.getDoubleControl(XPRS_FEASTOL), tol);
 }
 
+TEST_F(XpressFixtureLP, SetPrimalToleranceNotOverridenByMPSolverParameters) {
+  double tol = 1e-4;  // Choose a value different from kDefaultPrimalTolerance
+  std::string xpressParamString = "FEASTOL " + std::to_string(tol);
+  solver.SetSolverSpecificParametersAsString(xpressParamString);
+  solver.Solve();
+  EXPECT_EQ(getter.getDoubleControl(XPRS_FEASTOL), tol);
+}
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface_test.cc:775]: Add tests for other types of "controls" ? I see 4 in total
- String
- Double (OK)
- Integer
- Integer64

Comment (tbittar) [reply ortools/linear_solver/xpress_interface_test.cc:775]: The goal here is to test that parameters available as MPSolverParameters do not override the corresponding parameter set from specific parameters. And all relevant MPSolverParameters are tested

**Comment by flomnes** [comment]
It seems like you have a few conflicts, and also some changes that are not related to the PR title. Can you fix this @tbittar ?

**Review by pet-mit — APPROVED** [review]

**Comment by pet-mit** [comment]
ok to merge @flomnes  ?

**Review by flomnes — APPROVED** [review]

**Pull request merged**